### PR TITLE
Add RelayTask/SubTask Data/Service

### LINKED ISSUE #102: Task Data/Service
### Is this the right issue type?

- [x] Yes, I'm planning work for this project team.

### Summary

Create database entities and a service for `Task`

## Datamodel

`Task`
```
id: string
collection: string
created: datetime
completed: datetime? (when a task has been handled by Relay)
```

`SubTask`
```
id: string
owner: fk to SubNode
task: fk to Task
result?: string (present when a result has been received)
```

## Task Service methods

create
get (id)
list 
update

## SubTask Service Methods

create
update

### Acceptance Criteria

- [ ] Data model exists for task 
- [ ] Data model exists for subtask
- [ ] CRUD Service for `Task`
- [ ] CRUD Service for `SubTask`

### Tasks

- [x] New entity classes:
    - [x] Task
    - [x] SubTask
- [x] Register in Db Context
- [x] Migrations
- [ ] Services:
    - [x] TaskService
    - [ ] SubTaskService
- [ ] Register services 
- [ ] Unittests on services:
    - [x] TaskService (all methods)
    - [ ] SubTaskService (all methods)

### Confirm creation

- [x] This issue is ready

### PR DESCRIPTION
**Pull request by AndyRae**
## Pull Request Contents

| |
|-|
✨ Feature
🛠️ Repo maintenance

## Description

Adds the data model, and service for `RelayTask`, and `RelaySubTask`, with associated tests.
These 2 models provide the "state" model in Relay.

The entities are prepended with `Relay`, as just using `Task` is probably not a good idea for namespace reasons.

Includes:
- Task/SubTask entities
- Migrations
- Models
- Services - with useful methods
- Tests on the service methods
- `.vscode` Workspace, and settings for Pytest on Resolve

## Related Issues or other material

- [x] This PR relates to one or more issues, detailed below

- Closes #102 

## ✅ Added/updated tests?

- [x] This PR contains relevant tests
  - Or doesn't need to per the below explanation

- [x] I've completed all **actions** and **tasks** detailed in the PR Template

### CODE REVIEW

File: app/Hutch.Relay/Models/RelaySubTaskModel.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using Hutch.Relay.Data.Entities;
```
Comment (beforan) [inline app/Hutch.Relay/Models/RelaySubTaskModel.cs:1]: Why?

File: .vscode/hutch.code-workspace [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "name": "Resolve",
+      "path": "../app/resolve"
+    },
+    {
+      "name": "RquestBridge",
+      "path": "../app/RquestBridge"
+    },
+    {
+      "name": "FiveSafes.Net",
+      "path": "../lib/react-client-app"
```
Comment (beforan) [inline .vscode/hutch.code-workspace:21]: Think (hope?) this path is incorrect

File: .vscode/hutch.code-workspace [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    {
+      "name": "RACKit",
+      "path": "../app/Hutch.RACKit"
```
Comment (beforan) [inline .vscode/hutch.code-workspace:25]: Think (hope?) this path is incorrect

**Review by beforan — CHANGES_REQUESTED** [review]

File: .vscode/hutch.code-workspace (hunk 2 above)
Comment (AndyRae) [reply .vscode/hutch.code-workspace]: Yep - fixed

File: .vscode/hutch.code-workspace (hunk 3 above)
Comment (AndyRae) [reply .vscode/hutch.code-workspace]: Fixed

File: app/Hutch.Relay/Models/RelaySubTaskModel.cs (hunk 1 above)
Comment (AndyRae) [reply app/Hutch.Relay/Models/RelaySubTaskModel.cs]: I had a missing model - `RelaySubTask` needed `SubNodeModel`, this has been added.

**Review by beforan — APPROVED** [review]

**Review by vpnu — APPROVED** [review]

**Pull request merged**